### PR TITLE
chore: refactor login

### PR DESCRIPTION
**Pull request by markusahlstrand**
Moved code into the password flow so we can reuse it with the /co/authenticate endpoint

### CODE REVIEW

File: src/models/CustomError.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { HTTPException } from "hono/http-exception";
+import { StatusCode } from "hono/utils/http-status";
+
+export type HTTPExceptionCode =
+  | "EMAIL_NOT_VERIFIED"
+  | "INVALID_PASSWORD"
+  | "USER_NOT_FOUND";
+
+export type HTTPExceptionOptions = {
```
Comment (danielgent) [inline src/models/CustomError.ts:9]: Should the `HTTP` be camel-case like the `Http` in `XMLHttpRequest` or like the `XML`? :joy:  :facepalm: 

https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest

File: src/authentication-flows/password.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  const user = await getPrimaryUserByEmailAndProvider({
+    userAdapter: ctx.env.data.users,
+    tenant_id: client.tenant_id,
+    email,
+    provider: "auth2",
+  });
+
+  // route always returns success
+  if (!user) {
+    return ctx.html("We've just sent you an email to reset your password.");
```
Comment (danielgent) [inline src/authentication-flows/password.ts:34]: looks like you're bringing back old code. I think we need a test that snapshots universal login when requesting a password reset for a non-existent account (although not informing the user is odd)

**Review by danielgent — APPROVED** [review]

File: src/models/CustomError.ts (hunk 1 above)
Comment (markusahlstrand) [reply src/models/CustomError.ts:9]: Haha.. Yeah, the web isn't super consistent. I haven't really decided what to do.. Specially with stuff like XmlHttp, it kind of get harder to read with XMLHTTP, right? Almost leaning towards Http..

File: src/authentication-flows/password.ts (hunk 2 above)
Comment (markusahlstrand) [reply src/authentication-flows/password.ts:34]: Yeah, I changed the name of the file and this was fixed in the other version. It won't really matter as we don't use the response, but it's still wrong.. I'll fix it

**Pull request merged**